src/main/scala/mesosphere/marathon/storage/repository/Repositories.scala

```diff
@@ -53,6 +54,27 @@ trait GroupRepository {
   def podVersions(id: PathId): Source[OffsetDateTime, NotUsed]
 
   def podVersion(id: PathId, version: OffsetDateTime): Future[Option[PodDefinition]]
+
+  def runSpecVersion(id: PathId, version: OffsetDateTime)(implicit executionContext: ExecutionContext): Future[Option[RunSpec]] = {
```

Comment:
It's kinda funny that we ourselves don't know if the instance belongs to an app or a pod and have to check both.